### ISSUE TITLE
pgagroal connections are in Graceful state since a long time

### ISSUE THREAD

**Issue by sheshjee** [agroal/pgagroal#451]
[postgres@dbm ~]$ pgagroal-cli status
Status:              Running
Active connections:  876
Total connections:   883
Max connections:     1000
[postgres@dbm ~]$ pgagroal-cli details
Status:              Running
Active connections:  876
Total connections:   883
Max connections:     1000
---------------------
Server:              primary
Host:                localhost
Port:                5432
State:               Primary
---------------------
Connection    0:     Graceful        2024-03-05 17:47:30 939245 12     mydb postgres
Connection    1:     Active          2024-05-21 14:30:05 987972 974    mydb postgres
Connection    2:     Graceful        2024-03-05 21:20:41 2123523 14     mydb postgres
Connection    3:     Graceful        2024-03-05 17:19:23 776934 13     mydb postgres
Connection    4:     Graceful        2024-03-05 18:41:07 1260190 18     mydb postgres
Connection    5:     Graceful        2024-03-05 16:58:24 663693 15     mydb postgres
Connection    6:     Graceful        2024-03-06 15:05:27 2609519 16     mydb postgres
Connection    7:     Graceful        2024-03-06 11:20:41 1769345 17     mydb postgres
Connection    8:     Graceful        2024-03-05 19:36:38 1562277 19     mydb postgres
Connection    9:     Graceful        2024-03-06 11:36:44 1797066 20     mydb postgres
Connection   10:     Graceful        2024-03-05 20:36:51 1873176 21     mydb postgres
Connection   11:     Graceful        2024-03-06 11:36:43 1796850 22     mydb postgres
Connection   12:     Graceful        2024-03-06 12:08:50 1952867 23     mydb postgres
Connection   13:     Graceful        2024-03-06 17:38:13 2899712 24     mydb postgres
Connection   14:     Graceful        2024-03-06 17:38:13 2899647 25     mydb postgres
Connection   15:     Graceful        2024-03-06 11:57:51 1890399 33     mydb postgres
Connection   16:     Graceful        2024-03-06 17:36:32 2895999 26     mydb postgres
Connection   17:     Graceful        2024-03-06 12:08:50 1952866 27     mydb postgres
Connection   18:     Graceful        2024-03-09 08:53:07 3416234 28     mydb postgres
Connection   19:     Graceful        2024-03-06 11:57:51 1890398 29     mydb postgres
Connection   20:     Graceful        2024-03-06 17:38:13 2899711 31     mydb postgres

Please suggest me why connections are in graceful state since a long time and is there any specific parameter in pgagroal.conf to free these connections??

**Comment by fluca1978** [comment]
Seems pgagroal is not able to terminate the connections. 
Have you tried to run `pgagroal-cli flush gracefully`?
Which version are you running?

**Comment by sheshjee** [comment]
Yes I tried and it removed the connections. But the issue is why
connections are not being closed by itself.

pgagroal - 1.5

**Comment by fluca1978** [comment]
It's hard to tell why `gracefully` connections are not closed in your case.
And as already stated, version `1.5` is behind the current release, so while I'm not sure this could fix the problem, upgrading could make easier to debug.
Moreover, understanding what PostgreSQL sees as connections could help. My suspect is that you have something preventing connections to be automatically closed, could it be an application?

**Comment by sheshjee** [comment]
Yes I think the same as client sends request by an application code to my
Database.

**Comment by fluca1978** [comment]
I'm closing this issue since there has been no update in months, and it is referring to an old version of the pooler.
@sheshjee in case you are still experiencing the problem, please reopen the issue and provide more details about it.
